src/core/page-parser.ts

```diff
@@ -1,67 +1,96 @@
-import {JSDOM} from "jsdom";
-import {Request} from "express";
-import {CssSelectorDefinition, CssSelectorRegistry} from "./css-selector-registry";
+import {JSDOM} from 'jsdom';
+import {snakeCase} from 'lodash';
 // @ts-ignore
 import * as RegexTranslator from 'regex-translator';
+import axios from 'axios';
 
-import {snakeCase} from 'lodash';
+import {CssSelectorDefinition, CssSelectorRegistry} from "./css-selector-registry";
 
-const axios = require('axios').default;
+export type Language = 'en' | 'de' | 'ja' | 'fr';
+export type RequestFunction = (url: string, headers?: Record<string, string>) => Promise<string>;
+
+const lodestoneBaseUrls: Record<Language, string> = {
+    'en': 'https://na.finalfantasyxiv.com/lodestone',
+    'de': 'https://de.finalfantasyxiv.com/lodestone',
+    'ja': 'https://jp.finalfantasyxiv.com/lodestone',
+    'fr': 'https://fr.finalfantasyxiv.com/lodestone',
+};
```

Comment:
I'm not sure what the language support would have to offer, since we're getting a lot of things in english so we can convert to ids that can then be used to get the name of the element in any language (including ko/zh) using the ID itself (and it's lighter for the reply, too)

Reply:
I didn't see any of the "convert to ID" in the functionality of the codebase yet, sorry!
I assumed it would be more light-weight in terms of requests to the Lodestone (which is the main limiting factor for this lib AFAIK) to request data in the language needed, rather than requesting IDs and then later resolving them with a new request.
An alternative would be to resolve the IDs locally, but that would involve some kind of DB, maybe saved locally alongside the lib, which would produce a lot of complexity in code and keeping the DB up to date.
What would be your alternative approach to language support?

Reply:
I don't think Nodestone should provide language support at all, using IDs will make sure that the data can be used in any language, as it becomes the consumer's responsability to convert to string for their own display.

It also makes the response payload way lighter by not including strings.

Also, every XIVAPI entity that has a name has the name in 4 languages, having nodestone only provide one language based on the selected language would push people into making 4 requests just for localization purpose, which would be a bad thing overall.